src/connector/src/parser/parquet_parser.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::sync::Arc;
+
+use arrow_array_iceberg::RecordBatch;
+use futures_async_stream::try_stream;
+use risingwave_common::array::arrow::IcebergArrowConvert;
```

Comment:
How's iceberg related here?

Reply:
Here we just use the conversion method in `IcebergArrowConvert`. `IcebergArrowConvert` provides a series of methods to convert from `arrow_type` to `data_type`. I guess it was named `IcebergArrowConvert` because it was only used in iceberg before, and we may rename it more generally, like `ArrowConvert` in later pr.
For the `arrow_array_iceberg` and `arrow_array`, they are just different versions of the same lib arrow_array(the same goes for `arrow_schema` etc.). I suggest not to add iceberg in the name. 
The versions will be unified and upgraded to 52 in the next PR.